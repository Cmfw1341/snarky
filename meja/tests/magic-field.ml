module Impl = Snarky.Snark.Make (Snarky.Backends.Mnt4.Default)
open Impl

type field_pair = field * field

type field_pair_alias = field_pair

type record_containing_field = {a: int * (int * field)}

type 'a with_param = unit * 'a

type field_param = field with_param

type variant_containing_field = A of field | B

type _ variant_with_field_param =
  | A : int variant_with_field_param
  | B : field variant_with_field_param

let add3 (add : Field.t -> Field.t -> Field.t) =
  add
    (add
       (Field.constant (Field.Constant.of_string "1"))
       (Field.constant (Field.Constant.of_string "2")))
    (Field.constant (Field.Constant.of_string "3"))
