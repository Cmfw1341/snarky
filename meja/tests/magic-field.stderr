File "tests/magic-field.meja", line 22, characters 8-11:
Error: Incompatable types _ and int -> int -> int:
       Cannot unify field and int
